packages/components/src/Chips/InternalChipDismissible/hooks/useInternalChipDismissibleInput.tsx

```diff
@@ -0,0 +1,195 @@
+import React, {
+  ChangeEvent,
+  KeyboardEvent,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import { v1 as uuidV1 } from "uuid";
+import { debounce } from "lodash";
+import {
+  ChipDismissibleInputOptionProps,
+  ChipDismissibleInputProps,
+} from "../InternalChipDismissibleTypes";
+import { Icon } from "../../../Icon";
+import { ChipProps } from "../../Chip";
+
+const menuId = uuidV1();
+
+export function useInternalChipDismissibleInput({
+  options,
+  isLoadingMore = false,
+  onEmptyBackspace,
+  onCustomOptionSelect,
+  onOptionSelect,
+  onSearch,
+}: ChipDismissibleInputProps) {
+  const [allOptions, setAllOptions] = useState<
+    ChipDismissibleInputOptionProps[]
+  >([]);
+  const [searchValue, setSearchValue] = useState("");
+  const [menuOpen, setMenuOpen] = useState(false);
+  const [activeIndex, setActiveIndex] = useState(0);
+  const [shouldCancelBlur, setShouldCancelBlur] = useState(false);
+  const canAddCustomOption =
+    onCustomOptionSelect !== undefined && !isLoadingMore;
+  const maxOptionIndex = allOptions.length - 1;
+
+  useEffect(() => {
+    setAllOptions(generateOptions(options, searchValue, canAddCustomOption));
+  }, [options]);
+
+  const computed = {
+    menuId,
+    inputRef: useRef<HTMLInputElement>(null), // eslint-disable-line no-null/no-null
+    activeOption: allOptions[activeIndex],
+    hasAvailableOptions: allOptions.length > 0,
+    nextOptionIndex: activeIndex < maxOptionIndex ? activeIndex + 1 : 0,
+    previousOptionIndex: activeIndex > 0 ? activeIndex - 1 : maxOptionIndex,
+  };
+
+  const actions = {
+    generateDescendantId: (index: number) => `${computed.menuId}-${index}`,
+
+    handleReset: () => {
+      setActiveIndex(activeIndex === 0 ? activeIndex : activeIndex - 1);
+      setSearchValue("");
+    },
+
+    handleOpenMenu: () => setMenuOpen(true),
+
+    handleCloseMenu: () => {
+      setMenuOpen(false);
+      setActiveIndex(0);
+    },
+
+    handleCancelBlur: () => setShouldCancelBlur(true),
```

Comment:
Wondering if these ones would read better if they were called `cancelBlur` for example?  Maybe it's a naming convention I don't know about either though!

Reply:
i'd keep it as handle. Keep it consistent with the other functions that gets called in an `on` props

Reply:
I see, it's a consistency thing okay!